components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/ApiResourceManager.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.api.resource.mgt;
+
+import org.wso2.carbon.identity.api.resource.mgt.model.ApiResourceSearchResult;
+import org.wso2.carbon.identity.application.common.model.ApiResource;
+import org.wso2.carbon.identity.application.common.model.Scope;
+
+import java.util.List;
+
+/**
+ * Api Resource Manager Interface.
+ */
+public interface ApiResourceManager {
+
+    /**
+     * Get API resources.
+     *
+     * @param filter       Filter expression.
+     * @param limit        Limit value.
+     * @param after        After value.
+     * @param before       Before value.
+     * @param tenantDomain Tenant domain.
+     * @return API resource search result.
+     * @throws ApiResourceManagementException API resource management exception.
```

Comment:
```suggestion
     * @throws ApiResourceManagementException If error occurred while getting API Resources.
```

Reply:
Check for similar places

Reply:
https://github.com/wso2/carbon-identity-framework/pull/4854/commits/8dd16447d6000412e3ff2dd1b2a9ed78249f2d13